core/arch/arm/sm/sm.c

```diff
@@ -39,6 +39,11 @@ bool sm_from_nsec(struct sm_ctx *ctx)
 {
 	uint32_t *nsec_r0 = (uint32_t *)(&ctx->nsec.r0);
 
+#if defined(PLATFORM_ti)
```

Comment:
Please use a better config variable, `CFG_SM_PLATFORM_HANDLER` for instance.

If `CFG_SM_PLATFORM_HANDLER` only is `y` when the platform actually provides the `sm_platform_handler()` then we wouldn't need the `__weak` function below. Instead it could be a static inline (in `<sm/sm.h>`) that always returns `true`, making this `#if defined` redundant.

One advantage with this approach is that a platform provided `sm_platform_handler()` isn't silently ignored if `CFG_SM_PLATFORM_HANDLER` isn't `y`.